invenio_records_permissions/permissions/base.py

```diff
@@ -6,99 +6,108 @@
+# |    delete   |   x  |     x    |               |
+# |-------------|------|----------|---------------|
+#
 
-class _PermissionConfig(object):
 
-    # Deny all by default
-    can_create = [Deny]
-    can_list = [Deny]
-    can_read = [Deny]
-    can_update = [Deny]
-    can_delete = [Deny]
-
-    @classmethod
-    def get_permission_list(cls, action):
-        if action == 'create':
-            return cls.can_create
-        elif action == 'list':
-            return cls.can_list
-        elif action == 'read':
-            return cls.can_read
-        elif action == 'update':
-            return cls.can_update
-        elif action == 'delete':
-            return cls.can_delete
-
-        current_app.logger.error("Unkown action {action}.".format(
-            action=action))
-        return []
+class BasePermissionPolicy(Permission):
+    """
+    BasePermissionPolicy to inherit from.
 
+    The class defines the overall policy and the instance encapsulates the
+    permissions for an *action* *over* a set of objects.
 
-# Where can a property be used?
-#
-# |    Action   | need | excludes | query_filter |
-# |-------------|------|----------|--------------|
-# |    create   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     list    |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |     read    |   x  |     x    |       x      |
-# |-------------|------|----------|--------------|
-# | read files  |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    update   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-# |    delete   |   x  |     x    |              |
-# |-------------|------|----------|--------------|
-#
+    TODO: Recognize a PermissionPolicy class in other modules instead of
+          individual factory functions to lessen the configuration burden.
+    """
 
+    # Deny all by default
+    can_list = [Deny()]
+    can_create = [Deny()]
+    can_read = [Deny()]
+    can_update = [Deny()]
+    can_delete = [Deny()]
+
+    def __init__(self, action, **over):
+        """Constructor."""
+        super(BasePermissionPolicy, self).__init__()
+        self.action = action
+        self.over = over
 
-class BasePermission(Permission):
+    @property
+    def generators(self):
+        """List of Needs generators for self.action."""
+        action = self.action
+        cls = self.__class__
+        if hasattr(cls, 'can_' + action):
+            return getattr(cls, 'can_' + action)
+
+        current_app.logger.error("Unknown action {action}.".format(
+            action=action))
 
-    def __init__(self, config, action):
-        super(BasePermission, self).__init__()
-        self.config = config
-        self.permission_list = self.config.get_permission_list(action)
+        # Question: should this default to [Deny()] or []?
+        return [Deny()]
```

Comment:
Since the configuration above returns `Deny`, and this would be an `Unknown` action, I would prefer to be safe than sorry and `Deny` it.

Reply:
If we change the ones above to just `[]`, this should also be `[]`, right?